liota/dccs/iotcc.py

```diff
@@ -109,29 +110,24 @@ def register(self, entity_obj):
             def on_response(msg):
                 try:
                     log.debug("Received msg: {0}".format(msg))
-                    if msg != "":
-                        json_msg = json.loads(msg)
-                        log.debug("Processed msg: {0}".format(json_msg["type"]))
-                        if json_msg["type"] == "create_or_find_resource_response":
-                            if json_msg["body"]["uuid"] != "null" and json_msg["body"]["id"] == entity_obj.entity_id:
-                                log.info("FOUND RESOURCE: {0}".format(json_msg["body"]["uuid"]))
-                                self.reg_entity_id = json_msg["body"]["uuid"]
-                            else:
-                                # TODO : Remove this code if required
-                                log.info("Waiting for resource creation")
-                                self.comms.send(json.dumps(
-                                    self._registration(self.next_id(), entity_obj.entity_id, entity_obj.name,
-                                                       entity_obj.entity_type)))
-                                on_response(self.recv_msg_queue.get())
+                    json_msg = json.loads(msg)
```

Comment:
Can you please elaborate, what exactly was causing the problem and how it got resolved? I didn't understand it from the changes.